CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/SyntaxNodeFile.swift

```diff
@@ -0,0 +1,359 @@
+              leadingTrivia: childDoc ?? [],
+              modifiers: [DeclModifierSyntax(name: .keyword(.public))],
+              name: PatternSyntax("\(raw: child.swiftName)"),
+              type: TypeAnnotationSyntax(type: type)) {
+                AccessorDeclSyntax(accessorKind: .keyword(.get)) {
+                  if child.isOptional {
+                    StmtSyntax("return data.child(at: \(raw: index), parent: Syntax(self)).map(\(raw: childType).init)")
+                  } else {
+                    StmtSyntax("return \(raw: childType)(data.child(at: \(raw: index), parent: Syntax(self))!)")
+                  }
+                }
+
+                AccessorDeclSyntax("""
+                  set(value) {
+                    self = \(raw: node.name)(data.replacingChild(at: \(raw: index), with: value\(raw: child.isOptional ? "?" : "").raw, arena: SyntaxArena()))
+                  }
+                  """)
+              }
+
+            // ===============
+            // Adding children
+            // ===============
+            // We don't currently support adding elements to a specific unexpected collection.
+            // If needed, this could be added in the future, but for now withUnexpected should be sufficient.
+            if let childNode = SYNTAX_NODE_MAP[child.syntaxKind], childNode.isSyntaxCollection, !child.isUnexpectedNodes {
+              let childElt = child.collectionElementName!
+              let childEltType = childNode.collectionElementType.syntaxBaseName
+
+              DeclSyntax("""
+                /// Adds the provided `\(raw: childElt)` to the node's `\(raw: child.swiftName)`
+                /// collection.
+                /// - param element: The new `\(raw: childElt)` to add to the node's
+                ///                  `\(raw: child.swiftName)` collection.
+                /// - returns: A copy of the receiver with the provided `\(raw: childElt)`
+                ///            appended to its `\(raw: child.swiftName)` collection.
+                public func add\(raw: childElt)(_ element: \(raw: childEltType)) -> \(raw: node.name) {
+                  var collection: RawSyntax
+                  let arena = SyntaxArena()
+                  if let col = raw.layoutView!.children[\(raw: index)] {
+                    collection = col.layoutView!.appending(element.raw, arena: arena)
+                  } else {
+                    collection = RawSyntax.makeLayout(kind: SyntaxKind.\(raw: childNode.swiftSyntaxKind),
+                                                      from: [element.raw], arena: arena)
+                  }
+                  let newData = data.replacingChild(at: \(raw: index), with: collection, arena: arena)
+                  return \(raw: node.name)(newData)
+                }
+                """)
+            }
+          }
+
+          try! VariableDeclSyntax("public static var structure: SyntaxNodeStructure") {
+            ReturnStmtSyntax(
```

Comment:
Similar to a case above I think this can be simplified as

```swift
let layout = ArrayExprSyntax {
  for child in node.children {
    ArrayElementSyntax(
      expression: ExprSyntax(#"\Self.\#(raw: child.swiftName)"#))
  }
}
StmtSyntax("return .layout(\(layout))")
```